perceval/components/processor.py

```diff
@@ -53,27 +53,57 @@ class Processor(AProcessor):
         >>> p = Processor("SLOS", BS() // PS() // BS())
 
     :param source: the Source used by the processor (defaults to perfect source)
+    :param noise: a NoiseModel containing noise parameters (defaults to no noise)
+                  Note: source and noise are mutually exclusive
     :param name: a textual name for the processor (defaults to "Local processor")
     """
-    def __init__(self, backend: Union[ABackend, str], m_circuit: Union[int, ACircuit] = None, source: Source = Source(),
-                 name: str = None):
+    def __init__(self, backend: Union[ABackend, str], m_circuit: Union[int, ACircuit] = None, source: Source = None,
+                 noise: NoiseModel = None, name: str = "Local processor"):
         super().__init__()
-        self._source = source
-        self.name = "Local processor" if name is None else name
+        self._init_backend(backend)
+        self._init_circuit(m_circuit)
+        self._init_noise(source, noise)
+        self.name = name
+        self._inputs_map: Union[SVDistribution, None] = None
+        self._simulator = None
 
+    def _init_noise(self, source: Source, noise: NoiseModel):
+        # Default = perfect simulation
+        self._phase_quantization = 0
+        self._source = Source()
```

Comment:
You can initialize _source to None, and then at the end of the method do:
```
if self_source is None:
    self._source = Source()
```
instead of initializing an object you may not need ?